_inc/client/rest-api/index.js

```diff
@@ -258,3 +258,7 @@ function checkStatus( response ) {
 		throw error;
 	} );
 }
+
+function catchJsonParseError( e ) {
+	throw new Error( `Couldn't understand Jetpack's REST API response (${ e.name })` );
```

Comment:
Since this is going to face the user, let's rephrase please and use transatable strings here?